### PR TITLE
CATROID-1169 Refactored AddUserDataToUserDefinedBrickFragment and AddUserDefinedBrickFragment from Java to Kotlin

### PR DESCRIPTION
**Pull request by dorianpercic**
Jira ticket: https://jira.catrob.at/browse/CATROID-1169
Refactored AddUserDataToUserDefinedBrickFragment and AddUserDefinedBrickFragment from Java to Kotlin. Took null safety into account.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDataToUserDefinedBrickFragment.kt [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        }
+        nextItem = menu.findItem(R.id.next)
+        nextItem?.isVisible = true
+        nextItem?.isEnabled = true
+        super.onPrepareOptionsMenu(menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        if (item.itemId == R.id.next) {
+            val fragmentManager = fragmentManager
```
Comment (ChriMoser) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDataToUserDefinedBrickFragment.kt:176]: In the old file the keyboard was hidden at this position, why not here aswell?

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDefinedBrickFragment.kt [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        confirmItem?.isEnabled = true
+        super.onPrepareOptionsMenu(menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        if (item.itemId == R.id.confirm) {
+            val currentSprite = ProjectManager.getInstance().currentSprite
+            val brickIsEmpty = userDefinedBrick?.isEmpty
+
+            if (brickIsEmpty == true) {
```
Comment (ChriMoser) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDefinedBrickFragment.kt:137]: is "== true" needed?

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDefinedBrickFragment.kt [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+                userDefinedBrick?.addLabel("")
+            }
+
+            if (currentSprite.doesUserBrickAlreadyExist(userDefinedBrick)) {
+                ToastUtil.showErrorWithColor(
+                    context,
+                    R.string.brick_user_defined_already_exists,
+                    Color.RED
+                )
+                if (brickIsEmpty == true) {
```
Comment (ChriMoser) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDefinedBrickFragment.kt:147]: is "== true" needed?

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDataToUserDefinedBrickFragment.kt (hunk 1 above)
Comment (ChriMoser) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDataToUserDefinedBrickFragment.kt:176]: In the old file the keyboard was hidden at this position, why not here aswell?

**Review by ChriMoser — CHANGES_REQUESTED** [review]
Looks good, but some minor changes/questions to resolve.

Please also fix the Static Analysis warnings from Jenkins!

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDefinedBrickFragment.kt [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.menu_confirm_userdefined, menu)
+        super.onCreateOptionsMenu(menu, inflater)
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        for (index in 0..menu.size()) {
+            menu.getItem(index).isVisible = false
+        }
```
Comment (Walcho1125) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDefinedBrickFragment.kt:122]: This also causes 
```
AndroidRuntime: FATAL EXCEPTION: main
    Process: org.catrobat.catroid, PID: 17448
    java.lang.IndexOutOfBoundsException: Index: 12, Size: 12
        at java.util.ArrayList.get(ArrayList.java:437)
        at androidx.appcompat.view.menu.MenuBuilder.getItem(MenuBuilder.java:752)
        at org.catrobat.catroid.ui.fragment.AddUserDefinedBrickFragment.onPrepareOptionsMenu(AddUserDefinedBrickFragment.kt:121)
        at androidx.fragment.app.Fragment.performPrepareOptionsMenu(Fragment.java:2820)
        at androidx.fragment.app.FragmentManager.dispatchPrepareOptionsMenu(FragmentManager.java:2703)
        at androidx.fragment.app.FragmentController.dispatchPrepareOptionsMenu(FragmentController.java:399)
        at androidx.fragment.app.FragmentActivity.onPreparePanel(FragmentActivity.java:488)
        at androidx.appcompat.view.WindowCallbackWrapper.onPreparePanel(WindowCallbackWrapper.java:99)
        at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.onPreparePanel(AppCompatDelegateImpl.java:3097)
        at androidx.appcompat.view.WindowCallbackWrapper.onPreparePanel(WindowCallbackWrapper.java:99)
        at androidx.appcompat.app.ToolbarActionBar$ToolbarCallbackWrapper.onPreparePanel(ToolbarActionBar.java:523)
        at androidx.appcompat.app.ToolbarActionBar.populateOptionsMenu(ToolbarActionBar.java:457)
        at androidx.appcompat.app.ToolbarActionBar$1.run(ToolbarActionBar.java:57)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:966)
        at android.view.Choreographer.doCallbacks(Choreographer.java:790)
        at android.view.Choreographer.doFrame(Choreographer.java:721)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:951)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```
```suggestion
        for (index in 0 until menu.size()) {
            menu.getItem(index).isVisible = false
        }
```

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDataToUserDefinedBrickFragment.kt [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.menu_next, menu)
+        super.onCreateOptionsMenu(menu, inflater)
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        for (index in 0..menu.size()) {
+            menu.getItem(index).isVisible = false
+        }
```
Comment (Walcho1125) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDataToUserDefinedBrickFragment.kt:170]: This results in:
```
AndroidRuntime: FATAL EXCEPTION: main
    Process: org.catrobat.catroid, PID: 17448
    java.lang.IndexOutOfBoundsException: Index: 12, Size: 12
        at java.util.ArrayList.get(ArrayList.java:437)
        at androidx.appcompat.view.menu.MenuBuilder.getItem(MenuBuilder.java:752)
        at org.catrobat.catroid.ui.fragment.AddUserDefinedBrickFragment.onPrepareOptionsMenu(AddUserDefinedBrickFragment.kt:121)
        at androidx.fragment.app.Fragment.performPrepareOptionsMenu(Fragment.java:2820)
        at androidx.fragment.app.FragmentManager.dispatchPrepareOptionsMenu(FragmentManager.java:2703)
        at androidx.fragment.app.FragmentController.dispatchPrepareOptionsMenu(FragmentController.java:399)
        at androidx.fragment.app.FragmentActivity.onPreparePanel(FragmentActivity.java:488)
        at androidx.appcompat.view.WindowCallbackWrapper.onPreparePanel(WindowCallbackWrapper.java:99)
        at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.onPreparePanel(AppCompatDelegateImpl.java:3097)
        at androidx.appcompat.view.WindowCallbackWrapper.onPreparePanel(WindowCallbackWrapper.java:99)
        at androidx.appcompat.app.ToolbarActionBar$ToolbarCallbackWrapper.onPreparePanel(ToolbarActionBar.java:523)
        at androidx.appcompat.app.ToolbarActionBar.populateOptionsMenu(ToolbarActionBar.java:457)
        at androidx.appcompat.app.ToolbarActionBar$1.run(ToolbarActionBar.java:57)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:966)
        at android.view.Choreographer.doCallbacks(Choreographer.java:790)
        at android.view.Choreographer.doFrame(Choreographer.java:721)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:951)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```
caused by an off-by-one error, instead use:
```suggestion
        for (index in 0 until menu.size()) {
            menu.getItem(index).isVisible = false
        }
```

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDataToUserDefinedBrickFragment.kt [hunk 6]
```diff
@@ -0,0 +1,219 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2021 The Catrobat Team
```
Comment (Walcho1125) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDataToUserDefinedBrickFragment.kt:3]: please update in every file
```suggestion
 * Copyright (C) 2010-2022 The Catrobat Team
```

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDefinedBrickFragment.kt [hunk 7]
```diff
@@ -0,0 +1,231 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2021 The Catrobat Team
```
Comment (Walcho1125) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserDefinedBrickFragment.kt:3]:
```suggestion
 * Copyright (C) 2010-2022 The Catrobat Team
```

**Review by Walcho1125 — CHANGES_REQUESTED** [review]
Please look at the remarks as the app crashes because of an off-by-one error and update the copyright line in every file

**Review by Walcho1125 — APPROVED** [review]
Thanks!

**Pull request merged**